src/single_node_consensus.rs

```diff
@@ -0,0 +1,169 @@
+    utils::modules::Module,
+};
+use anyhow::Result;
+use bincode::{Decode, Encode};
+
+bus_client! {
+struct SingleNodeConsensusBusClient {
+    sender(ConsensusEvent),
+    sender(SignedByValidator<MempoolNetMessage>),
+    receiver(ConsensusCommand),
+    receiver(MempoolEvent),
+    receiver(MempoolNetMessage),
+    receiver(SignedByValidator<ConsensusNetMessage>),
+    receiver(OutboundMessage),
+    receiver(Query<QueryConsensusInfo, ConsensusInfo>),
+}
+}
+
+#[derive(Encode, Decode, Default)]
+struct SingleNodeConsensusStore {
+    cars: Vec<Car>,
+}
+
+pub struct SingleNodeConsensus {
+    bus: SingleNodeConsensusBusClient,
+    car_proposal_signer: SharedBlstCrypto,
+    crypto: SharedBlstCrypto,
+    store: SingleNodeConsensusStore,
+    file: Option<PathBuf>,
+}
+
+/// The `SingleNodeConsensus` module listens to and sends the same messages as the `Consensus` module.
+/// However, there are differences in behavior:
+/// - It does not perform any consensus.
+/// - It creates a fake validator that automatically votes for any `CarProposal` it receives.
+/// - For every CarProposal received, it saves it automatically as a `Car`.
+/// - When Mempool sends a new `Cut`, it is able to retrieve the associated `Car`s and create a new `CommitCut`
+impl Module for SingleNodeConsensus {
+    fn name() -> &'static str {
+        "SingleNodeConsensus"
+    }
+
+    type Context = SharedRunContext;
+
+    async fn build(ctx: Self::Context) -> Result<Self> {
+        let file = ctx
+            .common
+            .config
+            .data_directory
+            .clone()
+            .join("consensus_single_node.bin");
+
+        let store: SingleNodeConsensusStore = Self::load_from_disk_or_default(file.as_path());
+
+        let bus = SingleNodeConsensusBusClient::new_from_bus(ctx.common.bus.new_handle()).await;
+        Ok(SingleNodeConsensus {
+            bus,
+            car_proposal_signer: Arc::new(BlstCrypto::new(
+                "single_node_car_proposal_signer".to_owned(),
+            )),
+            crypto: ctx.node.crypto.clone(),
+            store,
+            file: Some(file),
+        })
+    }
+
+    fn run(&mut self) -> impl futures::Future<Output = Result<()>> + Send {
+        self.start()
+    }
+}
+
+impl SingleNodeConsensus {
+    async fn start(&mut self) -> Result<()> {
+        let validators = vec![
+            self.car_proposal_signer.validator_pubkey().clone(),
+            self.crypto.validator_pubkey().clone(),
+        ];
+        let new_bonded_validators = vec![self.crypto.validator_pubkey().clone()];
+
+        if self.store.cars.is_empty() {
+            // Send new cut with two validators. The Node itself and the 'car_proposal_signer' which is here just to Vote for new CarProposal.
+            self.bus.send(ConsensusEvent::CommitCut {
+                validators,
+                new_bonded_validators,
+                cut: Cut::default(),
+            })?;
+        }
+
```

Comment:
ce serait chouette d'avoir un log 
```suggestion
info!("Waiting for a first transaction before creating blocks...");
```

Car c'est de fait ce qu'on attend, et quand on start la node single, ça hang et on comprends pas ce qui se passe... alors que tout va bien, juste on attend une première transaction :)